core/controllers/reader.py

```diff
@@ -1040,3 +1040,31 @@ def get(self):
             'question_dicts': question_dicts
         })
         self.render_json(self.values)
+
+
+class LearnerAnswerInfoSubmissionHandler(base.BaseHandler):
+    """Handles the learner answer details submission."""
+
+    REQUIRE_PAYLOAD_CSRF_CHECK = False
+
+    @acl_decorators.can_play_exploration
+    def post(self, exploration_id):
+        """"Handles the POST requests. Stores the answer details submitted
+        by the learner.
+        """
+
+        if not constants.ENABLE_SOLICIT_ANSWER_DETAILS_FEATURE:
+            raise self.PageNotFoundException
+
+        state_name = self.payload.get('state_name')
+        interaction_id = self.payload.get('interaction_id')
+        answer = self.payload.get('answer')
+        answer_details = self.payload.get('answer_details')
+        entity_type = feconf.ENTITY_TYPE_EXPLORATION
```

Comment:
Ditto about having it generic.

Reply:
There are different ACL layers for editing question and exploration, if I will make the function generic then how will I combine the ACL layers? I think to make them different for both types i.e. question and exploration will be good to manage. What do you suggest @aks681 & @DubeySandeep ?

Reply:
Ok, it's fine to have this be specific to explorations then. It is in reader.py after all.

Reply:
> There are different ACL layers for editing question and exploration

This one has `can_play_exploration`, instead I think we can have can_submit_answer, right?

Also, I think we can find a way to write ACL for 'get' as well*

Reply:
I think ACL layer for can_play_exploration is fine for submission , because if you are playing the exploration then you can submit the answer,  and how one can only play the exploration not submit any answer? What do you suggest? @aks681 @DubeySandeep

Reply:
I was asking to have can_submit_answer acl so that we can make this handler work for both exploration and questions.

Reply:
@anubhavsinha98 You should find a reasonable way to make this generic.

Reply:
Just FYI, try to look at how permissions are handled for suggestions to see how you can make this generic (and use other ACL functions if you want to). (Ref https://github.com/oppia/oppia/pull/5170)